core/src/consensus.rs

```diff
@@ -75,7 +75,8 @@ pub fn relative_min_window_density(b1: &MinaConsensusState, b2: &MinaConsensusSt
 
     let projected_window = {
         // Compute shift count
-        let shift_count = (max_slot - global_slot(b1) - 1).clamp(0, SUB_WINDOWS_PER_WINDOW);
+        let shift_count = 
+            (max_slot.wrapping_sub(global_slot(b1) + 1)).clamp(0, SUB_WINDOWS_PER_WINDOW);
```

Comment:
@olton thank you! this should use `saturating_sub` instead.

Reply:
This version would provide the intended behavior:

```rust
let shift_count = max_slot
    .saturating_sub(global_slot(b1) + 1)
    .clamp(0, SUB_WINDOWS_PER_WINDOW);
```

as per [the spec](https://github.com/MinaProtocol/mina/tree/develop/docs/specs/consensus#5412-relative-minimum-window-density):

```
let shift_count = min(
    max(max_slot - B1.curr_global_slot - 1, 0), // same as `saturating_sub`
    sub_windows_per_window
)
```